iter0 is complete and meets all specs

kctrlv opened the pull request:
Great work on iter0! This will serve as a solid non-turdlike foundation for the rest of our code.

kctrlv committed:
added master test for sales_engine

kctrlv committed:
got a test to pass

kctrlv committed:
added gitignore

kctrlv committed:
top-level salesengine base requirements

kctrlv committed:
item class in progress

kctrlv committed:
item class can init with correct attributes

kctrlv committed:
item class solid

kctrlv committed:
itemrepo initialize, all solid

kctrlv committed:
find by id, name, description done

kctrlv committed:
made simple test data for itemsrepo

kctrlv committed:
itemrepo all methods done and tested

kctrlv committed:
salesengine works with itemrepo

kctrlv committed:
used correct form for csv file map row method (to_a)

kctrlv committed:
merchant class base tests

kctrlv committed:
merchant class solid

kctrlv committed:
various cleaning and add simple merchants datatest

kctrlv committed:
merchantrepo, itemrepo, merchant, item, all ok

kctrlv committed:
completed iter0

danbroadbent commented:
Talked to another group about the sales engine class.  I think we should discuss whether it needs to be initialized.

kctrlv commented:
It appeared that we had no problems but we learned that our implementation of the price data was operating under the assumption that unit_price was the price in dollars. After learning that the unit_data was the price in cents, we updated the code, fixed our sample csv data, and got both our tests and the spec harness to pass. Will push updates to iter0

kctrlv committed:
made some refactors, fixed sample data related to unit_price, verified passed spec_harness0, and our tests

danbroadbent commented:
mrspec passes, spec harness passes, good to go
